python-frontend/src/main/java/org/sonar/python/semantic/v2/types/TrivialTypeInferenceVisitor.java

```diff
@@ -153,11 +157,31 @@ public void visitNone(NoneExpression noneExpression) {
 
   @Override
   public void visitListLiteral(ListLiteral listLiteral) {
-    ModuleType builtins = this.projectLevelTypeTable.getModule();
+    var builtins = this.projectLevelTypeTable.getModule();
     scan(listLiteral.elements());
-    List<PythonType> pythonTypes = listLiteral.elements().expressions().stream().map(Expression::typeV2).distinct().toList();
+
+    var candidateTypes = listLiteral.elements()
+      .expressions()
+      .stream()
+      .map(Expression::typeV2)
+      .distinct()
+      .toList();
+
+    var elementsType = candidateTypes.size() == 1 ? candidateTypes.get(0) : candidateTypes
+      .stream()
+      .reduce(new UnionType(new HashSet<>()), (c1, c2) -> {
+        if (Stream.of(c1, c2).anyMatch(UnknownType.class::isInstance)) {
+          return PythonType.UNKNOWN;
+        } else {
+          ((UnionType) c1).candidates().add(c2);
+          return c1;
+        }
+      });
+
+    var attributes = new ArrayList<PythonType>();
+    attributes.add(elementsType);
```

Comment:
Could this be a one-liner? Or, you don't want to use something like `List#of` to keep the list mutable? Genuine question.

Reply:
I wanted to keep it mutable